### PR TITLE
Change exchange

### PR DESCRIPTION
**Pull request by Krestol**

### CODE REVIEW

File: tests/StateObserver.py [hunk 1]
```diff
@@ -10,31 +10,60 @@
...
-        self.totalProfit = InternalRevenueServiceWrapper.GetTotalReward(assetId, addr)
         self.successfulOperations = RequestHolderWrapper.GetSuccessfulOperationsCount(addr)
         self.failedOperations = RequestHolderWrapper.GetFailedOperationsCount(addr)
         self.isBlocked = OnyxPayWrapper.IsBlockedUser(addr)
-        self.fiatAmount = ExchangeWrapper.GetFiatAmount(addr, assetId)
 
+    def PrintState(self):
+        print(self.addr)
+        print("Balances:")
+        for i, asset in enumerate(self.balances):
```
Comment (arsonaut) [inline tests/StateObserver.py]: i and enumerate() aren't used here, also it's better to use .items()
```suggestion
        for asset, balance in self.balances.items():
```

File: tests/StateObserver.py [hunk 2]
```diff
@@ -10,31 +10,60 @@
...
         self.successfulOperations = RequestHolderWrapper.GetSuccessfulOperationsCount(addr)
         self.failedOperations = RequestHolderWrapper.GetFailedOperationsCount(addr)
         self.isBlocked = OnyxPayWrapper.IsBlockedUser(addr)
-        self.fiatAmount = ExchangeWrapper.GetFiatAmount(addr, assetId)
 
+    def PrintState(self):
+        print(self.addr)
+        print("Balances:")
+        for i, asset in enumerate(self.balances):  
+            print("{}: {}".format(asset, self.balances[asset]))
```
Comment (arsonaut) [inline tests/StateObserver.py]:
```suggestion
            print("{}: {}".format(asset, balance))
```

File: tests/StateObserver.py [hunk 3]
```diff
@@ -43,11 +72,16 @@ def AddTarget(self, addr):
...
     def GetState(self, label, addr) -> State:
         return self.states[label][addr]
 
     def GetGeneralState(self, label):
-        return self.states[label]['general']
\ No newline at end of file
+        return self.states[label]['general']
+
+    def PrintStates(self):
+        for label, state in enumerate(self.states):
```
Comment (arsonaut) [inline tests/StateObserver.py]:
```suggestion
        for state in self.states.values():     
            for stateObj in state.values():  
                stateObj.PrintState()
```

File: tests/OnyxPayUser.py [hunk 4]
```diff
@@ -100,6 +104,29 @@ def AddReferral(self, referrer):
         ReferralServiceWrapper.AddReferral(self.address, referrer, self.account)
         SdkUtils.WaitNextBlock()
 
+    def Exchange(self, assetToBuy, assetToSell, amountToBuy):
+        self.stateObserver.GrapState("before_exchange")
```
[the remark this answers is not in the document]
Comment (Krestol) [reply tests/OnyxPayUser.py]: Yes, I'll add it.

File: tests/StateObserver.py (hunk 1 above)
Comment (Krestol) [reply tests/StateObserver.py]: +

File: tests/StateObserver.py (hunk 2 above)
Comment (Krestol) [reply tests/StateObserver.py]: +

File: tests/StateObserver.py (hunk 3 above)
Comment (Krestol) [reply tests/StateObserver.py]: \+

**Pull request merged**